scaladoc/src/dotty/tools/scaladoc/site/SidebarParser.scala

```diff
@@ -6,31 +6,76 @@ import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.core.`type`.TypeReference;
 import collection.JavaConverters._
 import java.util.Optional
+import scala.beans._
 
 enum Sidebar:
-  case Category(title: Option[String], indexPath: Option[String], nested: List[Sidebar])
-  case Page(title: Option[String], pagePath: String)
+  case Root(index: Option[String], pages: List[Sidebar.Child])
+  case Category(
+    title: Option[String],
+    indexPath: Option[String],
+    nested: List[Sidebar.Child],
+    directory: Option[String]
+  )
+  case Page(title: Option[String], pagePath: String, hidden: Boolean)
 
 object Sidebar:
-  case class RawInput(var title: String, var page: String, var index: String, var subsection: JList[RawInput]):
-    def this() = this("", "", "", JList())
 
-    def setTitle(t: String) = this.title = t
-    def setPage(p: String) = this.page = p
-    def setIndex(i: String) = this.index = i
-    def setSubsection(s: JList[RawInput]) = this.subsection = s
+  type Child = Category | Page
+  case class RawRoot(var rootIndex: String, var pages: JList[RawInput]):
+    def this() = this("", JList())
 
-  type RawInnerTpe = JMap[String, JList[RawInput]]
-  private object RawTypeRef extends TypeReference[RawInnerTpe]
+    def setRootIndex(s: String) = rootIndex = s
+    def setPages(l: JList[RawInput]) = pages = l
 
-  private def toSidebar(r: RawInput): Sidebar = r match
-    case RawInput(title, page, index, subsection) if page.nonEmpty && index.isEmpty && subsection.isEmpty() || title == "Blog" =>
-      Sidebar.Page(Option.when(title.nonEmpty)(title), page)
-    case RawInput(title, page, index, subsection) if page.isEmpty && !subsection.isEmpty() =>
-      Sidebar.Category(Option.when(title.nonEmpty)(title), Option.when(index.nonEmpty)(index), subsection.asScala.map(toSidebar).toList)
+  case class RawInput(
+    @BeanProperty var title: String,
```

Comment:
Why do we need `@BeanProperty`?

Reply:
Previously, we had explicitly defined setters. I found the `BeanProperty` shorter and resistant to possible errors (changed types or sth). The reason is that we use Jackson library that can only deserialize YAML to bean class.